### PR TITLE
Functional Code for Pointcloud Fusion with RealSense and Azure Kinect Cameras

### PR DESCRIPTION
**Pull request by TheoMadaBoy**
Title: Functional Code for Pointcloud Fusion with RealSense and Azure Kinect Cameras

Description: This pull request introduces a functional implementation for the fusion of pointclouds from two cameras: a RealSense and an Azure Kinect. While the core C++ code and its associated configuration (e.g., CMakeLists.txt) are fully operational, the launch files and tf_transform setup remain to be refined to improve calibration accuracy.

Changes:

    Added a C++ module for real-time pointcloud fusion.
    Integrated ROS 2 support for RealSense and Azure Kinect cameras.
    Updated CMakeLists.txt to include the necessary dependencies.
    Provided initial tf_transform setup for camera calibration (still subject to improvement).

Testing:

    Verified the C++ implementation with sample pointcloud data for both cameras.
    Conducted initial real-world testing to confirm successful fusion and alignment.
    Ensured compatibility with the ROS 2 framework and dependent libraries.

Next Steps:

    Further refine the tf_transform values in the launch files to enhance calibration accuracy.
    Explore additional testing scenarios to validate robustness.

Dependencies:

    ROS 2 environment (compatible version).
    Proper drivers for RealSense and Azure Kinect cameras.
    Pre-calibrated physical setup for both cameras.

References:

    Base branch: [main](https://github.com/Lab-CORO/pointcloud_fusion)
    Development branch: [dev](https://github.com/Lab-CORO/pointcloud_fusion/tree/dev)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "files.associations": {
```
Comment (will-44) [inline .vscode/settings.json]: Il faut exclure le fichier vscode avec le gitignore

File: .vscode/settings.json
Comment (saxtot) [inline .vscode/settings.json:1]: This type of file shouldn't be included in commits.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,32 +1,92 @@
...
+
+
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+endif()
+
+# ament_export_include_directories(include)
+# ament_export_targets(${PROJECT_NAME}_node HAS_LIBRARY_TARGET)
+# ament_export_dependencies(${dependencies_pkgs})
```
Comment (saxtot) [inline CMakeLists.txt]: Remove if not necessary

File: README.md [hunk 3]
```diff
@@ -1 +1,48 @@
...
+
+This repository contains a C++ package for fusing point clouds from two different cameras using ROS 2. The package also supports 2D images. It is designed to work with RealSense 2 and Azure Kinect cameras.
+
+# Prerequisites
+
+Before getting started, make sure you have the following:
+
+1. ROS 2 (compatible version with your dependencies)
+2. Drivers for RealSense 2 and Azure Kinect cameras
+3. Colcon for building ROS 2 packages
```
Comment (saxtot) [inline README.md]: Weird to specify a dependency to `colcon` in a `ROS2` project.

File: README.md [hunk 4]
```diff
@@ -1 +1,48 @@
...
+
+# Installation and Usage Steps
```
Comment (saxtot) [inline README.md:12]: Might want to specify that the intended usage is for the `curobo_ros` project and what setup (if any) it requires in that context.

File: README.md [hunk 5]
```diff
@@ -1 +1,48 @@
...
+```bash
+    ros2 launch pointcloud_fusion pointcloud_fusion.launch.py
+```
+# Features
+
+- Real-time fusion of point clouds from two cameras.
+- Support for RealSense 2 and Azure Kinect cameras.
+- Compatible with 2D images and 3D point clouds using ROS 2.
+
+# Contributors
```
Comment (saxtot) [inline README.md]: This header is totally useless. Your ChatGPT is spilling.

File: launch/pointcloud_fusion.launch.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            '0',  # roll  (en radians)
+            '0',  # pitch (en radians)
+            '0',  # yaw   (en radians)
+            'base_link',
+            'camera_base'
+        ],
+        output='log'
+    )
+
+    # pointcloud_fusion_node (inchangé)
```
Comment (saxtot) [inline launch/pointcloud_fusion.launch.py:42]: What is this comment supposed to mean?

File: launch/pointcloud_fusion.launch.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        package='tf2_ros',
+        executable='static_transform_publisher',
+        name='static_transform_publisher_link6_to_camera_link',
+        arguments=[
+            '-0.0171',  # x
+            '-0.0858',  # y
+            '0.0606',   # z
+            '-0.0287',  # roll  (en radians)
+            '-0.0315',  # pitch (en radians)
+            '-2.8281', # yaw   (en radians)
```
Comment (saxtot) [inline launch/pointcloud_fusion.launch.py]: Might want to leave a comment mentioning where these numbers come from.

File: README.md (hunk 4 above)
Comment (TheoMadaBoy) [reply README.md:12]: That is true but the code is not reliate to curobo function so it technicaly work for anything reliate to ROS2 and fusioning 2 pointcloud of camera sending depth image. So the context should be just after a calibration of two camera ?

Comment (saxtot) [reply README.md:12]: That is true, but specifying your intent can justify certain design decisions or restrictions. For example, the specific camera combination currently supported.

File: .vscode/settings.json
Comment (saxtot) [reply .vscode/settings.json:1]: Adding it to the `.gitignore` is good, but you now have to actually remove it from the project too. Either remove/edit its commit of origin or use the appropriate `git rm` command.

File: README.md [hunk 8]
```diff
@@ -1 +1,46 @@
...
+ros2 launch driver.launch.py
+```
+5. Launch the fusion package:
+```bash
+    ros2 launch pointcloud_fusion pointcloud_fusion.launch.py
+```
+# Features
+
+- Real-time fusion of point clouds from two cameras.
+- Support for RealSense 2 and Azure Kinect cameras.
```
Comment (will-44) [inline README.md:43]: Tested for ...

File: README.md [hunk 9]
```diff
@@ -1 +1,46 @@
...
+- Compatible with 2D images and 3D point clouds using ROS 2.
```
Comment (will-44) [inline README.md:44]: depth map and point clouds

Comment (will-44) [reply README.md:44]: actually, it's only compatible with depth map

**Review by will-44 — APPROVED** [review]

**Pull request merged**